File: 1241-decompress-run-length-encoded-list/decompress-run-length-encoded-list.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int n = nums.length;

        int arrSize = 0;
        for(int i = 0; i < n; i += 2){
            arrSize += nums[i];
        }

        int[] res = new int[arrSize];

        int startIdx = 0;
        for(int i = 0; i < n; i += 2){
            Arrays.fill(res, startIdx, startIdx + nums[i], nums[i + 1]);
            startIdx += nums[i];
        }

        return res;
    }
}

File: 1241-decompress-run-length-encoded-list/decompress-run-length-encoded-list.cpp
class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        
        int l = 0;
        int r = 1;
        vector<int> res;
        while(l< nums.size() && r<nums.size())
        {
            for(int j = 0; j<nums[l]; j++){
                  res.push_back(nums[r]);
            }
            l+=2;
            r+=2;
        }
        return res;
    }
};